### DISCUSSION TITLE
expr-lang playground shows incorrect "Explain" for expressions

### DISCUSSION THREAD

**Discussion by rchougule** [antonmedv/expr#706] [General]
expr-lang supports short-circuiting of expressions. Though, the playground seems to be depicting otherwise.

For e.g.,
1. true || functionCall()
2. functionCall() || true

In both these cases, the `functionCall()` is not invoked actually when tested. But when looked at playground, it seems to **Explain** in a way that it does.
- Incorrect **Explain** 
<img width="1922" alt="image" src="https://github.com/user-attachments/assets/b1cc8d44-9d08-4378-9258-db7fee811deb">

- Correct **Bytecode** though
<img width="1922" alt="image" src="https://github.com/user-attachments/assets/9c9635a6-10aa-40cf-a81c-45d94ede4c6e">

**Comment by antonmedv** [comment]
Yes, this is "by design". In the explain more, Expr will evaluate both branches. I guess this is not intuitive behavior. I will turn on short-circuting by default in explain.

**Reply by rchougule** [reply]
Thanks a lot!

**Reply by rchougule** [reply]
Hi @antonmedv, just reaching out to check on when are we planning to enable this.
Thanks!

**Reply by antonmedv** [reply]
Hey, yes. This already implemented in Expr Pro. I have plans to deploy this to Expr Playground this week.
